Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -854,17 +860,85 @@ uint8_t AODProducerWorkflowDPL::getTRDPattern(const o2::trd::TrackTRD& track)
   return pattern;
 }
 
+// fill calo related tables (cells and calotrigger table)
+// currently hardcoded for EMCal, can be expanded for PHOS
+template <typename TCaloCells, typename TCaloTriggerRecord, typename TCaloCursor, typename TCaloTRGTableCursor>
+void AODProducerWorkflowDPL::fillCaloTable(const TCaloCells& calocells, const TCaloTriggerRecord& caloCellTRGR, const TCaloCursor& caloCellCursor,
+                                           const TCaloTRGTableCursor& caloCellTRGTableCursor, std::map<uint64_t, int>& bcsMap)
+{
+  uint64_t globalBC = 0;    // global BC ID
+  uint64_t globalBCRel = 0; // BC id reltive to minGlBC (from FIT)
+
+  // get cell belonging to an eveffillnt instead of timeframe
+  mEventHandler->reset();
+  mEventHandler->setCellData(calocells, caloCellTRGR);
+
+  // loop over events
+  for (int iev = 0; iev < mEventHandler->getNumberOfEvents(); iev++) {
+    o2::emcal::EventData inputEvent = mEventHandler->buildEvent(iev);
+    auto cellsInEvent = inputEvent.mCells;                  // get cells belonging to current event
+    auto interactionRecord = inputEvent.mInteractionRecord; // get interaction records belonging to current event
+
+    // Convert bc to global bc relative to min global BC found for all primary verteces in timeframe
+    // minGlBC and maxGlBC are set in findMinMaxBc(...)
+    globalBC = interactionRecord.toLong();
+
+    // check with Markus if globalBC ID is needed or globalBC - minGlBC
+    // in case of collision vertex what is used is
+    // uint64_t globalBC = std::round(tsTimeStamp / o2::constants::lhc::LHCBunchSpacingNS);
+    auto item = bcsMap.find(globalBC);
+    int bcID = -1;
+    if (item != bcsMap.end()) {
+      bcID = item->second;
+    } else {
+      throw CollisionIDNotFoundException(globalBC, bcsMap.size());
```

Comment:
as far as I know, we don't use exceptions in that code. It would be better to not divert here and simply use LOG(FATAL) or LOG(ERROR) ... which is much easier to treat.

Reply:
changed to your suggestion, following usage from other detectors